packages/remix-react/components.tsx

```diff
@@ -959,9 +959,10 @@ export function useFormAction(
   method: FormMethod = "get"
 ): string {
   let { id } = useRemixRouteContext();
+  let { manifest } = useRemixEntryContext();
   let path = useResolvedPath(action);
   let search = path.search;
-  let isIndexRoute = id.endsWith("/index");
+  let isIndexRoute = manifest.routes[id].index;
```

Comment:
Since `isIndexRoute` suggests a boolean value, this should be converted to a boolean value.

```suggestion
  let isIndexRoute = Boolean(manifest.routes[id].index);
```